app-modules/task/src/Policies/TaskPolicy.php

```diff
@@ -99,10 +99,11 @@ public function update(Authenticatable $authenticatable, Task $task): Response
             return Response::deny('You do not have permission to update this task.');
         }
 
-        return $authenticatable->canOrElse(
-            abilities: ["task.{$task->getKey()}.update"],
-            denyResponse: 'You do not have permission to update this task.'
-        );
+        if ($authenticatable->getKey() !== $task?->assigned_to && $authenticatable->getKey() !== $task?->created_by) {
```

Comment:
I don't think you need the `nullsafe` on `$task` in both instances here. There is not a scenario in this method where `$task` can be null